### DISCUSSION TITLE
The question about typing new creation options

### DISCUSSION THREAD

**Discussion by GuosXin** [vuejs/pinia#1875] [Help and Questions] [answered]
I try to adding a new option in a private pinia plugin by using Typescript.
And bundling it by using Webpack.
Then I install this package in my project by using npm.
And it has been added to the pinia instance with pinia.use().
Util now Everything is ok.
But when I difining a store to use the new option, the editor report a store option type error.
The option type expansion doesn't work in the package.
How can I solve this problem.

**Comment by posva** [comment, marked as answer]
If you don't share your plugin code you won't be able to get help 😉

**Reply by GuosXin** [reply]
Hi, here is a demo.
[https://github.com/GuosXin/pinia-plugin-demo](https://github.com/GuosXin/pinia-plugin-demo)

In the pinia-plugin-use-demo project, my pinia plugin can work. 
But the option type expansion doesn't work.
It seems that the TS configuration file in the pinia-plugin-use-demo project excludes node_modules folder.
But I don't want users to declare the option type expansion themselves.
How can I solve this problem in my plugin.

**Reply by posva** [reply]
Include the types at https://github.com/GuosXin/pinia-plugin-demo/blob/main/my-pinia-plugin/src/types.ts in your index.ts or make sure the final types.d.ts has a `export {}` to become a module and is included in types in tsconfig.json

**Reply by GuosXin** [reply]
Thank you, it work!
